jenkins-integration/lib/puppet/gatling/config.rb

```diff
@@ -18,7 +17,12 @@ def get_node_configs(scenario)
 end
 
 def get_modules(node_configs)
-  node_configs.reduce(Set.new) do |modules, node_config|
-    modules.merge(node_config['modules'])
-  end.to_a
+  result = Hash.new { |h, k| h[k] = Array.new }
+  node_configs.each do |node_config|
+    env = node_config['environment']
+    env = 'production' if (env.nil? || env.empty?)
+    result[env] += node_config['modules']
+  end
+  result.values.each &:uniq!
```

Comment:
Glad you understand the `Hash.new { ... }` part, but it was easier to remove duplicates with arrays+uniq! then with Set+to_array since the former can update in-place and the latter can't, meaning I would need more code to iterate and convert to arrays (and I don't want to return Set values since they're not that friendly to work with).